### PR TITLE
OCM-5542 | feat: add `--client` argument to `rosa version` command

### PR DESCRIPTION
**Pull request by jfchevrette**
https://issues.redhat.com/browse/OCM-5542

In some restricted environments in might not be desirable to have the version command do a network call

This adds a `--client` argument to `rosa version`. The naming of the argument align with other tools like kubectl/oc. In order to preserve current behavior this defaults to `false`.

### CODE REVIEW

File: cmd/version/cmd.go
Comment (ciaranRoche) [inline cmd/version/cmd.go:43]: Is this something we want to expose to our users? 

I am wondering if we want to use `MarkHidden` for now, to unblock your work, and we can check with BU if this is something we want to expose.

Comment (jfchevrette) [reply cmd/version/cmd.go:43]: I don't see any harm or risk in letting a user print a CLI tool version while explicitly disabling an unneeded external network call. Can you explain?

Comment (ciaranRoche) [reply cmd/version/cmd.go:43]: typically if we are not ready to support a new flag, we MarkHidden from the user. Functionality is still there. So I am wondering are we better in this case to do the same to unblock your effort. And if the need or ask from customers for this ability, we we can un-hide it.

Comment (gdbranco) [reply cmd/version/cmd.go:43]: I'd say there might definitely be customers that use it within an environment without connection. I see no harm in moving this through. Your call :)

**Review by ciaranRoche — APPROVED** [review]
/lgtm

**Pull request merged**
